### DISCUSSION TITLE
Raspi Pico and  STK8BA58 ±2g/±4g/±8g, 3-axis linear accelerometer, with digital output (I2C)

### DISCUSSION THREAD

**Discussion by Bernd54Albrecht** [micropython/micropython#12382] [Using MicroPython] [answered]
How do I initialize and query the registers of the accelerometer? No library/module available. I2C-Addr = 0x18
Examples: RANGESEL=Reg 0x0F  how set to 0000 0011 for 2g?
or how to read XOUT1 = Reg 0x02 for LSB, XOUT2 = Reg 0x03 for MSB
and so on. Thanks

**Comment by uGinn-R** [comment, marked as answer]
All the answers can be found here https://docs.micropython.org/en/latest/library/machine.I2C.html#machine-softi2c
Just some quick example of how it should be:

```
from machine import Pin, I2C

DEV_ADDR = 0x18
RANGESEL = 0x0F
XOUT1 = 0x02
XOUT2 = 0x03

i2c = I2C(scl=Pin(YOUR_SCL_PIN_NUMBER), sda=Pin(YOUR_SDA_PIN_NUMBER), freq=100000) # i2c init
i2c.writeto_mem(DEV_ADDR, RANGESEL, b'\x3') # selecting 2g. Hex "0x3" is the same as binary "0011"
xout1_reply = i2c.readfrom_mem(DEV_ADDR, XOUT1, 1) # read from device DEV_ADDR by XOUT1 memory address 1 byte. Same for XOUT2
```

**Reply by Bernd54Albrecht** [reply]
Thanks, I read the docs. But it simply does not work. Instantiation of I2C requires a 0 as first argument on the Raspberry Pi Pico - done.
i2c.scan() works fine (Hex address is 0x18)
However, (line 47)  i2c.writeto_mem(DEV_ADDR, RANGESEL, b'\x3') # selecting 2g. Hex "0x3" is the same as binary "0011" results in 
Traceback (most recent call last):
  File "<stdin>", line 47
SyntaxError: invalid syntax

And 
xout1_reply = i2c.readfrom_mem(DEV_ADDR, XOUT1, 1) # read from device DEV_ADDR by XOUT1 memory address 1 byte. Same for XOUT2
has a different reply everytime, although the sensor is not moving at all.

**Reply by uGinn-R** [reply]
Post the whole code with i2c operations you use. 

> Instantiation of I2C requires a 0 as first argument on the Raspberry Pi Pico
According to the sources first argument 0 means that you use default scl/sda pins (9 and 8).

> However, (line 47) i2c.writeto_mem(DEV_ADDR, RANGESEL, b'\x3') # selecting 2g. Hex "0x3" is the same as binary "0011" results in
Traceback (most recent call last):
File "", line 47
SyntaxError: invalid syntax

try send b'\x03' instead of b'\x3'
